libs/accounts/recovery-phone/src/lib/recovery-phone.repository.ts

```diff
@@ -1,18 +1,46 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-import { AccountDatabase } from '@fxa/shared/db/mysql/account';
+import { AccountDatabase, JsonValue } from '@fxa/shared/db/mysql/account';
 import { RecoveryPhone } from './recovery-phone.types';
 
+// Temporary helper function? Types are all over the place for lookupData,
+// should just be a JSON object or string that we parse into JSON?
+function isLookupDataObject(
+  val: JsonValue
+): val is { nationalFormat?: string } {
+  return (
+    typeof val === 'object' &&
+    !Array.isArray(val) &&
+    val?.['nationalFormat'] !== undefined
+  );
+}
+
 export async function getConfirmedPhoneNumber(
   db: AccountDatabase,
   uid: Buffer
-): Promise<{ uid: Buffer; phoneNumber: string } | undefined> {
-  return db
+): Promise<
+  { uid: Buffer; phoneNumber: string; nationalFormat?: string } | undefined
+> {
+  const row = await db
     .selectFrom('recoveryPhones')
     .where('uid', '=', uid)
-    .select(['uid', 'phoneNumber'])
+    .select(['uid', 'phoneNumber', 'lookupData'])
     .executeTakeFirst();
+
+  if (!row) {
+    return undefined;
+  }
+  const { uid: userId, phoneNumber, lookupData } = row;
+  const nationalFormat = isLookupDataObject(lookupData)
```

Comment:
This seems a lot simpler to me:

`const nationalFormat = lookupData?.['nationalFormat']`

We are at the DB layer here, so there might be some casting going on here. We know what we store is either null or a JsonObject, and if it's object it should contain nationalFormat, but might not. Either way the type guard doesn't do much for me, and I like the cleanliness of optimistic resolution. Just my two cents.

Reply:
Yeah this was meant to be a thing to clean up for sure. I had tried that, and I get this:

```
Element implicitly has an 'any' type because expression of type '"nationalFormat"' can't be used to index type 'string | number | boolean | JsonArray | JsonObject'.
  Property 'nationalFormat' does not exist on type 'string | number | boolean | JsonArray | JsonObject'.
```

I wasn't sure if there was a better way to pull JSON out? But I can just go with
`const nationalFormat = (lookupData as { nationalFormat?: string })?.nationalFormat` which works. 👍